libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/native/spm/SwiftPackageExtension.kt

```diff
@@ -0,0 +1,380 @@
+                }
+            }
+        }
+    }
+
+    // TODO: add fun with ranges
+    // TODO: add Package Dependency Requirement
+    class Dependencies {
+        var dependencies = mutableListOf<Package>()
+
+        fun `package`(url: String, version: String) {
+            val dependency = Package(url, version)
+            dependencies.add(dependency)
+        }
+
+        fun `package`(path: String) {
+            val dependency = Package(path)
+            dependencies.add(dependency)
+        }
+
+        data class Package(
+            @Input val url: String,
+            @Input @Optional var version: String? = null
+        )
+    }
+
+    class Targets {
+        // TODO: так и оставить как 3 списка?
+        var regulars = mutableListOf<Target>()
+        var executables = mutableListOf<Target>()
+        var tests = mutableListOf<Target>()
+
+        var libraries = mutableListOf<SystemLibrary>()
+        var binaries = mutableListOf<BinaryTarget>()
+
+        fun target(name: String, configure: Target.() -> Unit) {
+            val target = Target(name).apply(configure)
+            regulars.add(target)
+        }
+
+        fun executableTarget(name: String, configure: Target.() -> Unit) {
+            val target = Target(name).apply(configure)
+            executables.add(target)
+        }
+
+        fun testTarget(name: String, configure: Target.() -> Unit) {
+            val target = Target(name).apply(configure)
+            tests.add(target)
+        }
+
+        fun systemLibrary(name: String, configure: SystemLibrary.() -> Unit) {
+            val library = SystemLibrary(name).apply(configure)
+            libraries.add(library)
+        }
+
+        fun binaryTarget(name: String, url: String, checksum: String) {
+            val binaryTarget = BinaryTarget(name, url, checksum)
+            binaries.add(binaryTarget)
+        }
+
+        fun binaryTarget(name: String, path: String) {
+            val binaryTarget = BinaryTarget(name, path)
+            binaries.add(binaryTarget)
+        }
+
+        /** TODO
+         * Все target оказались одинаковыми по сигнатурам,
+         * кажется нет смысла разносить по разным классам
+         */
+        class Target(@Input val name: String) {
+            @Input
+            @Optional
+            var path: String? = null
+
+            @Input
+            @Optional
+            var publicHeadersPath: String? = null
+
+            @Nested
+            var dependenciesTarget = mutableListOf<TargetDependencies.Target>()
+
+            @Nested
+            var dependenciesProduct = mutableListOf<TargetDependencies.Product>()
+
+            @Nested
+            var excludePaths = mutableListOf<String>()
+
+            @Nested
+            @Optional
+            var sourceFiles: List<String>? = null
+
+            @Nested
+            @Optional
+            var resources: List<Resource>? = null
+
+            fun dependencies(configure: TargetDependencies.() -> Unit) {
```

Comment:
Functions with Kotlin lambdas parameters are visible only from `kts` scripts. To provide corresponding functionality to groovy scripts implement the overloading (see CocoapodsExtension.CocoapodsDependency#git(java.lang.String, groovy.lang.Closure<?>))